modules/gapi/samples/face_detection.cpp

```diff
@@ -0,0 +1,752 @@
+        tmcnnr_model_path,                // path to topology IR
+        weights_path(tmcnnr_model_path),  // path to weights
+        tmcnnr_target_dev,                // device specifier
+    }.cfgOutputLayers({ "conv5-2", "prob1" }).cfgInputLayers({ "data" });
+
+    // MTCNN Output detection network
+    std::vector<size_t> reshape_dims_48x48 = { 1, 3, 48, 48 };
+    auto mtcnno_net = cv::gapi::ie::Params<custom::MTCNNOutput>{
+        tmcnno_model_path,                // path to topology IR
+        weights_path(tmcnno_model_path),  // path to weights
+        tmcnno_target_dev,                // device specifier
+    }.cfgOutputLayers({ "conv6-2", "conv6-3", "prob1" }).cfgInputLayers({ "data" });
+
+    auto networks_mtcnn = cv::gapi::networks(mtcnnr_net, mtcnno_net);
+
+    // MTCNN Proposal detection network
+    for (int i = 0; i < pyramid_levels; ++i)
+    {
+        std::string net_id = "MTCNNProposal_" + std::to_string(level_size[i].width) + "x" + std::to_string(level_size[i].height);
+        std::cout << "mtcnnp_net " << net_id << std::endl;
+        std::vector<size_t> reshape_dims = { 1, 3, (size_t)level_size[i].width, (size_t)level_size[i].height };
+        cv::gapi::ie::Params<cv::gapi::Generic> mtcnnp_net{
+                    net_id,                           // tag
+                    tmcnnp_model_path,                // path to topology IR
+                    weights_path(tmcnnp_model_path),  // path to weights
+                    tmcnnp_target_dev,                // device specifier
+        };
+        mtcnnp_net.cfgInputReshape({ {"data", reshape_dims} });
+        networks_mtcnn += cv::gapi::networks(mtcnnp_net);
+    }
+
+    auto kernels_mtcnn = cv::gapi::kernels< custom::OCVBuildFaces
+        , custom::OCVRunNMS
+        , custom::OCVAccumulatePyramidOutputs
+        , custom::OCVApplyRegression
+        , custom::OCVBBoxesToSquares
+        , custom::OCVR_O_NetPreProcGetROIs
+        , custom::OCVRNetPostProc
+        , custom::OCVONetPostProc
+        , custom::OCVSwapFaces
+        , custom::OCVTranspose
+    >();
+    auto pipeline_mtcnn = graph_mtcnn.compileStreaming(cv::compile_args(networks_mtcnn, kernels_mtcnn));
+
+
+    std::cout << "Reading " << input_file_name << std::endl;
+    // Input stream
+    auto in_src = cv::gapi::wip::make_src<cv::gapi::wip::GCaptureSource>(input_file_name);
+
+    // Set the pipeline source & start the pipeline
+    pipeline_mtcnn.setSource(cv::gin(in_src));
+    pipeline_mtcnn.start();
+
+    // Declare the output data & run the processing loop
+    cv::TickMeter tm;
+    cv::Mat image;
+    std::vector<custom::Face> out_faces;
+
+    tm.start();
+    int frames = 0;
+    while (pipeline_mtcnn.pull(cv::gout(image, out_faces))) {
+        frames++;
+        std::cout << "Final Faces Size " << out_faces.size() << std::endl;
+        std::vector<rectPoints> data;
+        // show the image with faces in it
+        for (size_t i = 0; i < out_faces.size(); ++i) {
```

Comment:
Maybe the range-based for would work here (no need in `i`)

Reply:
Fixed